src/Idempotency.php

```diff
@@ -38,32 +38,26 @@ private static function getPrefix(): string
     public static function get(string $idempotencyKey): ?Response
     {
         $prefix = self::getPrefix();
-
-        if ($cached = self::getCache()->get("{$prefix}{$idempotencyKey}")) {
-            return new Response(...$cached);
-        }
-
-        return null;
+        return self::getCache()->get("{$prefix}{$idempotencyKey}");
     }
 
     /**
      * Add a response to the cache for the given key
      *
      * @param string $idempotencyKey
-     * @param \Illuminate\Http\Response $response
+     * @param \Symfony\Component\HttpFoundation\Response $response
      *
      * @return void
      */
     public static function add(string $idempotencyKey, Response $response): void
     {
-        $prefix = self::getPrefix();
+        if (property_exists($response, 'exception')) {
+            $response = clone $response;
```

Comment:
I clone the response here so that I don't strip the exception off the initial response. This will allow the response to keep the exception, incase it is needed after this is called.

Note, subsequent requests with the same idempotency key will return responses without the exception attached.

Reply:
Is there a reason why we cant just ignore responses with exceptions and not cache them?

Reply:
I could but then any endpoint that throws exceptions would have to manage the idempotency cache themselves.